conda/common/configuration.py

```diff
@@ -345,7 +344,12 @@ def make_raw_parameters_from_file(cls, filepath):
                 ruamel_yaml = yaml_load(fh)
             except ScannerError as err:
                 mark = err.problem_mark
-                raise LoadError("Invalid YAML", filepath, mark.line, mark.column)
+                raise ConfigurationLoadError(filepath,
+                                             "  reason: invalid yaml at line %s, column %s"
+                                             "" % (mark.line, mark.column))
+            except ReaderError as err:
+                raise ConfigurationLoadError(filepath, "  reason: invalid yaml at position %s"
+                                                       "" % err.position)
```

Comment:
Granted, `line`, `column` and `position` should always be integers, but in regards to https://github.com/conda/conda/issues/6759 / https://github.com/conda/conda/blob/4.6.0a2/conda/__init__.py#L48, I'd prefer every `CondaError` `message` to be a static string, e.g., either use
```python
class ConfigurationLoadError(ConfigurationError):
    def __init__(self, path, message_addition='', **kwargs):
[...]
                raise ConfigurationLoadError(filepath,
                                             "  reason: invalid yaml at line %(line)s, column %(column)s",
                                             line=mark.line, column=mark.column)
            except ReaderError as err:
                raise ConfigurationLoadError(filepath, "  reason: invalid yaml at position %(position)s",
                                                       position=err.position)
```
or
```python
class ConfigurationLoadError(ConfigurationError):
    def __init__(self, path, message_addition=''):
        message = "Unable to load configuration file.\n  path: %(path)s\n%(message_addition)s"
        super(ConfigurationLoadError, self).__init__(message, message_addition=message_addition, path=path)
```